pkg/strategy/production.go

```diff
@@ -120,9 +120,6 @@ func newProductionStrategy(ctx context.Context, jaeger *v1.Jaeger) S {
 	// prepare the deployments, which may get changed by the elasticsearch routine
 	cDep := collector.Get()
 	queryDep := inject.OAuthProxy(jaeger, query.Get())
-	if jaeger.Spec.Query.TracingEnabled == nil || *jaeger.Spec.Query.TracingEnabled == true {
```

Comment:
Why did you remove this `TracingEnabled` if?

Reply:
now the injection will be done purely via the deployment controller. So by removing this check I have to allow injection to `query` component in the deployment controller.

If this check is here the agent gets injected here with the current jaeger instance. I need to go throguh the deploymnt injection logic to allow injection via arbittrary jaeger instance.

Reply:
ok, make sense.